File: mqtt/src/main/java/org/sheedon/mqtt/DataConverter.java
package org.sheedon.mqtt;


import androidx.annotation.Nullable;

/**
 * 内容转化
 *
 * @Author: sheedon
 * @Email: sheedonsun@163.com
 * @Date: 2020/3/10 23:29
 */
public interface DataConverter<F, T> {
    T convert(F value);

    abstract class Factory {
        public @Nullable
        DataConverter<String, String> callbackNameConverter(String topic) {
            return null;
        }


    }
}

File: mqtt/src/main/kotlin/org/sheedon/mqtt/internal/DataConverter.kt
package org.sheedon.mqtt.internal

/**
 * 内容转换器
 *
 * @Author: sheedon
 * @Email: sheedonsun@163.com
 * @Date: 2022/1/8 4:55 下午
 */
interface DataConverter<T, F> {
    fun convert(value: T): F?
}